src/main/java/com/powsybl/openloadflow/ac/AbstractTransformerVoltageControlOuterLoop.java

```diff
@@ -48,4 +50,43 @@ public void cleanup(LfNetwork network) {
             branch.getVoltageControl().ifPresent(voltageControl -> branch.setVoltageControlEnabled(true));
         }
     }
+
+    protected void checkControl(LfNetwork network) {
+        List<LfBranch> controllerBranches = new ArrayList<>(1);
+        List<LfBranch> disabledBranches = new ArrayList<>(1);
+        for (LfBranch branch : network.getBranches()) {
+            if (!branch.isDisabled() && branch.isVoltageController() && branch.isVoltageControlEnabled()) {
+                controllerBranches.add(branch);
+            }
+            if (branch.isDisabled()) {
+                disabledBranches.add(branch);
+            }
+        }
+        for (LfBranch branch : controllerBranches) {
+            var connectivity = network.getConnectivity();
+            var voltageControl = branch.getVoltageControl().orElseThrow();
+            var controlledBus = voltageControl.getControlled();
+            // apply contingency (in case we are inside a security analysis)
+            for (LfBranch disabledBranch : disabledBranches) {
+                connectivity.cut(disabledBranch.getBus1(), disabledBranch.getBus2());
+            }
+            connectivity.cut(branch.getBus1(), branch.getBus2());
+            Set<LfBus> componentOnNotControlledSide = null;
+            if (controlledBus.equals(branch.getBus1())) {
+                componentOnNotControlledSide = connectivity.getConnectedComponent(branch.getBus2());
+            } else if (controlledBus.equals(branch.getBus2())) {
+                componentOnNotControlledSide = connectivity.getConnectedComponent(branch.getBus1());
+            } else {
+                // I don't know.
+            }
+            if (componentOnNotControlledSide != null) {
+                Optional<LfBus> generatorControlledBus = componentOnNotControlledSide.stream().filter(LfBus::isVoltageControlled).findAny();
```

Comment:
May be worth it to cache the result for this component, in order to not go over the same buses multiple times.

Reply:
The component depends on the contingency?

Reply:
Yes, I mean for this call of the function, not for successive calls:
if you have multiple transformers connected to the same component, we don't need to go over the buses of the component for each transformer.

Reply:
Actually there is a more sever issue than this very minor perf consideration:
we must not perform the connectivity cut and the component check in the same for loop!

We must :
1. cut all the transformers
2. check the component for each transformer

Otherwise, the check may find that first transformers are connected to a "voltage controlled" component, just because we have not yet disconnected other transformers.